prprocessor/__main__.py

```diff
@@ -282,13 +326,65 @@ async def update_redmine_on_issues(pull_request: Mapping, issues: Iterable[Issue
 
 @process_event_actions('pull_request', {'opened', 'ready_for_review', 'reopened', 'synchronize'})
 @process_webhook_payload
-async def on_pr_modified(*, pull_request: Mapping, **other) -> None:  # pylint: disable=unused-argument
-    await run_pull_request_check(pull_request)
+async def on_pr_modified(*, action: str, pull_request: Mapping, **_kw) -> None:
+    commits_valid_style = await run_pull_request_check(pull_request)
+
+    original_labels: set[str] = {label['name'] for label in pull_request['labels']}
+    labels_before = Label.load_labels(original_labels)
+    labels = labels_before.copy()
+
+    if action == 'opened':
+        labels.add(Label.NOT_YET_REVIEWED)
+
+    if action == 'synchronize' and Label.WAITING_ON_CONTRIBUTOR in labels:
+        labels.discard(Label.WAITING_ON_CONTRIBUTOR)
+        if pull_request['review_comments'] > 0:
```

Comment:
Looks like this isn't correct to determine if there's a review: https://github.com/ekohl/gh-test-env/pull/7#pullrequestreview-1617641711 was one, but didn't trigger addition of the new label.